dev_guide/builds.adoc

```diff
@@ -1557,3 +1557,84 @@ Docker and Source builds set the following labels on output images:
 |*io.openshift.build.source-location*
 |Source URL for the build
 |===
+
+
+[[using-external-artifacts]]
+= Using External Artifacts During The Build
+
+In order to use external artifacts during the build you don't need any special
+support from OpenShift platform. You could download and use any artifacts
+as you want.
+
+For `*Source*` build strategy you need to put appropriate shell commands into
+the `assemble` script.
+
+.`.sti/bin/assemble` file
+====
+
+[source,bash]
+----
+#!/bin/sh -x
+wget http://code.jquery.com/jquery-2.1.4.min.js
+----
+====
+
+For `*Docker*` build strategy you need to modify `Dockerfile` and invoke shell
+commands with https://docs.docker.com/engine/reference/builder/#run[`RUN` instruction].
+
+.Excerpt of `Dockerfile`
+====
+
+[source]
+----
+RUN wget http://code.jquery.com/jquery-2.1.4.min.js
+----
```

Comment:
Also here, I'd try to make a more usable example. Can we layer on top of an existing image and add a new binary or other artifact?
Don't need to be "copy-pastable", but at least something closer to a legit use:

```Dockerfile
FROM base-image:tag
...
RUN curl -O https://my.artifact.repository.com/builds/latest/app.war && do-something with app.war
...
CMD ["java", "..."]
```